src/store/slices/selectedItemDetailsSlice.ts

```diff
@@ -0,0 +1,22 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { IBookDetails } from "../../interfaces";
+
+export interface SelectedItemDetailsState {
+  details: IBookDetails;
+}
+const initialState: SelectedItemDetailsState = {
+  details: {} as IBookDetails,
```

Comment:
```suggestion
  details: null,
```
если у тебя нет начального значения, то ты или сетишь null и потом туда пихаешь то, что с бэка пришло. или же расписываешь какую-то структуру по умолчанию. IBookDetails у тебя огроный, поэтому или описать необходимые поля, которые ты дальше используешь. типа 
```
{
   title: "",
   date: "",
}
```
(*пример от балды)
или же присвоить null. такие значения проще дальше в коде обрабатывать. если просто воткуть пустой объект как IBookDetails, то тебе придется проверять каждое поле на наличие. иначе можно ошибку словить в самом неожиданном месте